### ISSUE TITLE
Generator incompatible with Dio v.2.2.1

### ISSUE THREAD

**Issue by rfogar2** [trevorwang/retrofit.dart#53]
**Describe the bug**
After upgrading my Flutter project to 1.9.1 and running `flutter pub upgrade`, my retrofit generator began throwing the following error: 

```
[SEVERE] Failed to snapshot build script .dart_tool/build/entrypoint/build.dart.
This is likely caused by a misconfigured builder definition.
[SEVERE] ../Downloads/flutter/.pub-cache/hosted/pub.dartlang.org/retrofit_generator-0.6.3/lib/src/generator.dart:499:38: Error: Getter not found: 'UploadFileInfo'.      final uploadFileInfo = refer('$UploadFileInfo')                                     ^^^^^^^^^^^^^^../Downloads/flutter/.pub-cache/hosted/pub.dartlang.org/retrofit_generator-0.6.3/lib/src/generator.dart:499:38: Error: The getter 'UploadFileInfo' isn't defined for the class 'RetrofitGenerator'. - 'RetrofitGenerator' is from 'package:retrofit_generator/src/generator.dart' ('../Downloads/flutter/.pub-cache/hosted/pub.dartlang.org/retrofit_generator-0.6.3/lib/src/generator.dart').Try correcting the name to the name of an existing getter, or defining a getter or field named 'UploadFileInfo'.      final uploadFileInfo = refer('$UploadFileInfo')                                     ^^^^^^^^^^^^^^
pub failed (78)
```

After playing with a few dependencies which had been upgraded, I singled out Dio (2.1.16 -> 2.2.1) as being the culprit. By setting `dio: 2.1.16` instead of `dio: ^2.1.16` in my pubspec.yaml file, I was able to run the generator with no problems. 

I don't know how `refer('$UploadFileInfo')` works, but it seems like the Dio upgrade has broken compatibility with `refer`. 

**To Reproduce**
1. Use Dio v.2.2.1
2. Use retrofit and retrofit_generator v.0.6.3
3. Run `flutter pub run build_runner build --delete-conflicting-outputs`

**Desktop (please complete the following information):**
 - OS: iOS

**Comment by devkabiir** [comment]
Yes, dio released version 3.0.0 today. It has many breaking changes https://github.com/flutterchina/dio/blob/master/dio/CHANGELOG.md. Currently using `dio: 2.1.16` is advised.

**Comment by devkabiir** [comment]
Dio does not have changelog for 2.2.0 and 2.2.1. They should not have released those versions with breaking changes. Because 2.2.0 and 2.2.1 satisfy version constraint `^2.1.0` retrofit_generator fails. Any one who uses `dio: ^2.1.0` will also fail

**Comment by devkabiir** [comment]
Related https://github.com/flutterchina/dio/issues/464

**Comment by wendux** [comment]
I just released a 2.2.2 version, which is compatible with 2.1. This is a quick fix.
